cmd/fs-v1.go

```diff
@@ -517,13 +518,29 @@ func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.
 
 	var wlk *lock.LockedFile
 	if bucket != minioMetaBucket {
-		fsMetaPath := pathJoin(fs.fsPath, minioMetaBucket, bucketMetaPrefix, bucket, object, fsMetaJSONFile)
+		bucketMetaDir := pathJoin(fs.fsPath, minioMetaBucket, bucketMetaPrefix)
+		fsMetaPath := pathJoin(bucketMetaDir, bucket, object, fsMetaJSONFile)
 		wlk, err = fs.rwPool.Create(fsMetaPath)
 		if err != nil {
 			return ObjectInfo{}, toObjectErr(traceError(err), bucket, object)
 		}
 		// This close will allow for locks to be synchronized on `fs.json`.
 		defer wlk.Close()
+		defer func() {
+			// Remove meta file when PutObject encounters any error
+			if err != nil {
```

Comment:
@vadmeste, I agree that `err` is not initialized inside the function. The value of `err` used inside the anonymous function is deferred to time of calling. This allows for programming errors like shadowing to cause unexpected value of `err ` to be bound inside the anonymous function. The code in its current form is fine. I am only cautioning of the kind of bugs that could creep in. These situations can be avoided by having conventions similar to avoiding shadowing of variables. This comment is meant in that spirit.